### ISSUE TITLE
Tests fail after Installing OpenMDAO

### ISSUE THREAD

**Issue by jsrogan** [OpenMDAO/OpenMDAO#3171]
I am working with a partner on making a contribution to an open source project, and one of our professors recommendations was OpenMDAO. We tried to follow along with the README.md instructions, but for whatever reason after installation there were still failures for both myself and my partner. My issue is as follows:

`The following tests failed:
test_external_code_comp.py:TestExternalCodeCompFeature.test_main
test_gui.py:gen_gui_test_case.test_gen_gui
test_gui.py:n2_gui_test_case.test_n2_gui

Passed:  2951
Failed:  3
Skipped: 906

Ran 3860 tests using 1 processes
Wall clock time:   00:09:35.69`

And my system information is:
 ('Linux', '#1 SMP Fri Apr 2 22:23:49 UTC 2021') ('x86_64', 'x86_64') 3.8.10 (default, May 26 2023, 14:05:08)  [GCC 9.4.0]

I have also attached the commands I ran and their respective outputs as a .txt file outlining the installation process of OpenMDAO.
[build.txt](https://github.com/OpenMDAO/OpenMDAO/files/14828812/build.txt)

**Comment by swryan** [comment]
You don't show what the actual errors were, the GUI tests can be finicky...  If you want to share how these particular tests failed, we might be able to comment further, but you are probably fine with these errors.  Neither of these features is core functionality.

Also, just to note.. you cloned the repository but you actually installed the release package from the PyPi package repository:
```
jsrogan@MSI:~/eecs481/hw6$ git clone http://github.com/OpenMDAO/OpenMDAO
Cloning into 'OpenMDAO'...
warning: redirecting to https://github.com/OpenMDAO/OpenMDAO/
remote: Enumerating objects: 133712, done.
remote: Counting objects: 100% (3391/3391), done.
remote: Compressing objects: 100% (828/828), done.
remote: Total 133712 (delta 2669), reused 3227 (delta 2552), pack-reused 130321
Receiving objects: 100% (133712/133712), 61.08 MiB | 1.82 MiB/s, done.
Resolving deltas: 100% (108526/108526), done.
jsrogan@MSI:~/eecs481/hw6$ pip install openmdao
Collecting openmdao
  Downloading openmdao-3.31.1.tar.gz (5.8 MB)
     |████████████████████████████████| 5.8 MB 732 kB/s 
```
To install from the local file system, you should be careful with case sensitivity or cd into the OpenMDAO directory before doing the `pip install`. The README file should probably be updated to the following to prevent this confusion:
```
git clone http://github.com/OpenMDAO/OpenMDAO
cd OpenMDAO
pip install .
```

**Comment by jsrogan** [comment]
Running the suggested pip install change to see if that changes anything when running tests, the 3 failures I ran into seemed to stem from either this: 

  File "/home/jsrogan/.local/lib/python3.8/site-packages/openmdao/components/external_code_comp.py", line 215, in _execute_local
    raise ValueError("The command to be executed, '%s', "
ValueError: 'p' <class ParaboloidExternalCodeComp>: Error calling compute(), The command to be executed, 'python', cannot be found

Or this:

║ Host system is missing dependencies to run browsers.
║ Please install them with the following command:      
║     sudo playwright install-deps

**Comment by swryan** [comment]
The latter message definitely explains the gui test failures, since it requires browser support.  If you are able to install the playwright dependencies as suggested, that should fix it.  Can you open a browser on that system?  If you don't need to work with the N2 diagram or any of the other visualization tools right now, then it shouldn't be a problem.

The first error is a little more curious.. it implies that the `python` command is not in your path.  Does `python --version` command work for you?   How about the `openmdao --version` command?

Are you using a tool such as [conda](https://docs.conda.io/projects/conda/en/stable/user-guide/getting-started.html) to manage you Python environment.  It not, you should consider it.

**Comment by jsrogan** [comment]
I am using a local version of python in a WSL system. I use python3 [command] (eg. python3 --version) in order to run these things on my subsystem. I am not using conda (causes testing irregularities on the Autograder where we have to submit our code for class projects)

python3 --version returns Python 3.8.10
openmdao --version returns 3.31.2-dev

**Comment by swryan** [comment]
Ah okay.. that explains the other test failure, since you don't have the expected `python` command.

You should be good to get started.

**Comment by sahinbatu** [comment]
I am also working on this open source project for the same contribution to an open source project assignment for the same class. My python is set to python3 already, however, mine is also creating 3 errors. Here is what they are, you can see the assertion errors as well:
<img width="947" alt="Screenshot 2024-04-02 at 15 53 20" src="https://github.com/OpenMDAO/OpenMDAO/assets/144289966/b9508902-e191-4715-a8a1-935c01658453">
It is a very minor float error, maybe altering the tolerance will help fix it. If going for accuracy though, it might be an error within the code.
Additionally, I wasn't able to download in edit mode, which could be another error?L
<img width="662" alt="Screenshot 2024-04-02 at 16 05 25" src="https://github.com/OpenMDAO/OpenMDAO/assets/144289966/a2cb015c-e972-45dc-8f3e-7d0eee1fa7f7">
Also, I wanted to ask if I could be added to the Slack chat if one exists? Because it seems like two teams within this class (my team, which is only me sadly, and the people who started this issue thread) are working on this open-source project, it would be great if we could have some communication so that we don't work on the same tasks accidentally.
Let me know if I should create a new issue thread for these bugs that the tests showed by the way!

**Comment by jsrogan** [comment]
@sahinbatu please feel free to reach out to me through UofM channels, we can coordinate what issues my partner and I have already chosen (we made a stackoverflow post about it yesterday with a list)

**Comment by swryan** [comment]
> I am also working on this open source project for the same contribution to an open source project assignment for the same class. My python is set to python3 already, however, mine is also creating 3 errors. Here is what they are, you can see the assertion errors as well: It is a very minor float error, maybe altering the tolerance will help fix it. If going for accuracy though, it might be an error within the code. 

If you could share the details of your environment (OS version per the README, Numpy/Scipy versions), that might be helpful.  It does look like a numerical sensitivity issue though, and not a problem with your install.

> Additionally, I wasn't able to download in edit mode, which could be another error?: 

When you are in the OpenMDAO directory, as it appears you were, then the proper command is `pip install -e .[all]`.    This will look in the current working directory for the `pyproject.toml` file.  Per the error message, you were telling pip to look in `hw6/OpenMDAO/OpenMDAO` which doesn't exist.

**Comment by sahinbatu** [comment]
`pip install -e .[all]` worked, thank you very much.

Here are the details of the environment:

- OS version:

  ```
  ProductName:            macOS
  ProductVersion:         14.3
  BuildVersion:           23D56
  ```

- Python 3.11.7
- Numpy 1.26.4
- Scipy 1.12.0

If it is a numerical sensitivity issue, should I create another issue for this?

**Comment by swryan** [comment]
Yes, that's a good idea...   somebody that can replicate your environment may then be able to reproduce your results.

In the meantime though, I think you are fine to continue working.
